==22297== Memcheck, a memory error detector
==22297== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==22297== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==22297== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global
==22297== 
==22297== Conditional jump or move depends on uninitialised value(s)
==22297==    at 0x68B7BD: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x6812BC: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x680531: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x299CCC: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2655B8: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x57C134: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25EBA1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25BED3: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25D952: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x72D972: {{closure}} (rt.rs:52)
==22297==    by 0x72D972: do_call<closure-0,i32> (panicking.rs:348)
==22297==    by 0x72D972: try<i32,closure-0> (panicking.rs:325)
==22297==    by 0x72D972: catch_unwind<closure-0,i32> (panic.rs:394)
==22297==    by 0x72D972: std::rt::lang_start_internal (rt.rs:51)
==22297==    by 0x25D601: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==  Uninitialised value was created by a stack allocation
==22297==    at 0x299BE0: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 
==22297== HEAP SUMMARY:
==22297==     in use at exit: 30,384 bytes in 91 blocks
==22297==   total heap usage: 847 allocs, 756 frees, 147,860 bytes allocated
==22297== 
==22297== 16 bytes in 1 blocks are still reachable in loss record 1 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x6BF738: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x6C3F76: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x6CC0C0: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x6BEC85: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x5821ED: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x57BF10: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25C6CA: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25D952: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x72D972: {{closure}} (rt.rs:52)
==22297==    by 0x72D972: do_call<closure-0,i32> (panicking.rs:348)
==22297==    by 0x72D972: try<i32,closure-0> (panicking.rs:325)
==22297==    by 0x72D972: catch_unwind<closure-0,i32> (panic.rs:394)
==22297==    by 0x72D972: std::rt::lang_start_internal (rt.rs:51)
==22297==    by 0x25D601: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 32 bytes in 1 blocks are still reachable in loss record 2 of 31
==22297==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==22297==    by 0x4E3F165: dlsym (dlsym.c:70)
==22297==    by 0x73BF43: fetch (weak.rs:61)
==22297==    by 0x73BF43: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==22297==    by 0x73BF43: min_stack_size (thread.rs:449)
==22297==    by 0x73BF43: std::sys::unix::thread::Thread::new (thread.rs:49)
==22297==    by 0x2DD4C6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 40 bytes in 1 blocks are still reachable in loss record 3 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x6CB71B: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x6BF728: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x6C3F76: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x6CC0C0: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x6BEC85: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x5821ED: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x57BF10: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25C6CA: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25D952: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x72D972: {{closure}} (rt.rs:52)
==22297==    by 0x72D972: do_call<closure-0,i32> (panicking.rs:348)
==22297==    by 0x72D972: try<i32,closure-0> (panicking.rs:325)
==22297==    by 0x72D972: catch_unwind<closure-0,i32> (panic.rs:394)
==22297==    by 0x72D972: std::rt::lang_start_internal (rt.rs:51)
==22297== 
==22297== 64 bytes in 4 blocks are still reachable in loss record 4 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2E677C: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E7767: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0C06: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 64 bytes in 4 blocks are still reachable in loss record 5 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x73BE9C: alloc (alloc.rs:80)
==22297==    by 0x73BE9C: alloc (alloc.rs:174)
==22297==    by 0x73BE9C: exchange_malloc (alloc.rs:277)
==22297==    by 0x73BE9C: std::sys::unix::thread::Thread::new (thread.rs:44)
==22297==    by 0x2DD4C6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 72 bytes in 1 blocks are still reachable in loss record 6 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x642FFD: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x646C6B: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x286AA5: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x280D75: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x57C4EE: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x57AE63: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25BFFD: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25D952: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x72D972: {{closure}} (rt.rs:52)
==22297==    by 0x72D972: do_call<closure-0,i32> (panicking.rs:348)
==22297==    by 0x72D972: try<i32,closure-0> (panicking.rs:325)
==22297==    by 0x72D972: catch_unwind<closure-0,i32> (panic.rs:394)
==22297==    by 0x72D972: std::rt::lang_start_internal (rt.rs:51)
==22297==    by 0x25D601: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 128 bytes in 4 blocks are still reachable in loss record 7 of 31
==22297==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==22297==    by 0x7084CE: register_dtor (fast_thread_local.rs:33)
==22297==    by 0x7084CE: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==22297==    by 0x7084CE: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==22297==    by 0x72ACC9: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==22297==    by 0x72ACC9: __getit (local.rs:175)
==22297==    by 0x72ACC9: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==22297==    by 0x72ACC9: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==22297==    by 0x72ACC9: std::sys_common::thread_info::set (thread_info.rs:40)
==22297==    by 0x2DD64C: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x73C2B9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22297==    by 0x73C2B9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22297==    by 0x73C2B9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22297==    by 0x52516DA: start_thread (pthread_create.c:463)
==22297==    by 0x57A271E: clone (clone.S:95)
==22297== 
==22297== 128 bytes in 4 blocks are still reachable in loss record 8 of 31
==22297==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==22297==    by 0x2EEF3D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E6922: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DE3E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2079: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E014A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2D90: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DD68E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x73C2B9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22297==    by 0x73C2B9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22297==    by 0x73C2B9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22297==    by 0x52516DA: start_thread (pthread_create.c:463)
==22297==    by 0x57A271E: clone (clone.S:95)
==22297== 
==22297== 160 bytes in 4 blocks are still reachable in loss record 9 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2E48AC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E1C7B: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E7AE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2BB0: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0CAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 160 bytes in 4 blocks are still reachable in loss record 10 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2E48AC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E1CA6: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E7AE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2BB0: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0CAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 160 bytes in 4 blocks are still reachable in loss record 11 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2E48AC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E794B: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0D20: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 160 bytes in 4 blocks are still reachable in loss record 12 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x70986E: alloc (alloc.rs:80)
==22297==    by 0x70986E: alloc (alloc.rs:174)
==22297==    by 0x70986E: exchange_malloc (alloc.rs:277)
==22297==    by 0x70986E: new<()> (mutex.rs:223)
==22297==    by 0x70986E: std::thread::Thread::new (mod.rs:1149)
==22297==    by 0x2DD3B1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 160 bytes in 4 blocks are still reachable in loss record 13 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2DD3D3: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 192 bytes in 4 blocks are still reachable in loss record 14 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x726966: alloc (alloc.rs:80)
==22297==    by 0x726966: alloc (alloc.rs:174)
==22297==    by 0x726966: exchange_malloc (alloc.rs:277)
==22297==    by 0x726966: std::sync::condvar::Condvar::new (condvar.rs:130)
==22297==    by 0x2E1C86: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E7AE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2BB0: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0CAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 192 bytes in 4 blocks are still reachable in loss record 15 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x726966: alloc (alloc.rs:80)
==22297==    by 0x726966: alloc (alloc.rs:174)
==22297==    by 0x726966: exchange_malloc (alloc.rs:277)
==22297==    by 0x726966: std::sync::condvar::Condvar::new (condvar.rs:130)
==22297==    by 0x2E1CB1: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E7AE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2BB0: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0CAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 192 bytes in 4 blocks are still reachable in loss record 16 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x726966: alloc (alloc.rs:80)
==22297==    by 0x726966: alloc (alloc.rs:174)
==22297==    by 0x726966: exchange_malloc (alloc.rs:277)
==22297==    by 0x726966: std::sync::condvar::Condvar::new (condvar.rs:130)
==22297==    by 0x2E7951: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0D20: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 192 bytes in 4 blocks are still reachable in loss record 17 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x7098D2: alloc (alloc.rs:80)
==22297==    by 0x7098D2: alloc (alloc.rs:174)
==22297==    by 0x7098D2: exchange_malloc (alloc.rs:277)
==22297==    by 0x7098D2: new (condvar.rs:130)
==22297==    by 0x7098D2: std::thread::Thread::new (mod.rs:1150)
==22297==    by 0x2DD3B1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 320 bytes in 4 blocks are still reachable in loss record 18 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x709921: alloc (alloc.rs:80)
==22297==    by 0x709921: alloc (alloc.rs:174)
==22297==    by 0x709921: exchange_malloc (alloc.rs:277)
==22297==    by 0x709921: new<std::thread::Inner> (sync.rs:324)
==22297==    by 0x709921: std::thread::Thread::new (mod.rs:1145)
==22297==    by 0x2DD3B1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 384 bytes in 1 blocks are still reachable in loss record 19 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2E5DCC: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2B6C: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0CAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 416 bytes in 4 blocks are still reachable in loss record 20 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2DD468: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 512 bytes in 1 blocks are still reachable in loss record 21 of 31
==22297==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x72C3D9: aligned_malloc (alloc.rs:95)
==22297==    by 0x72C3D9: alloc (alloc.rs:22)
==22297==    by 0x72C3D9: __rdl_alloc (alloc.rs:304)
==22297==    by 0x2E5FE6: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0CDA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 640 bytes in 1 blocks are still reachable in loss record 22 of 31
==22297==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x72C3D9: aligned_malloc (alloc.rs:95)
==22297==    by 0x72C3D9: alloc (alloc.rs:22)
==22297==    by 0x72C3D9: __rdl_alloc (alloc.rs:304)
==22297==    by 0x2E0F16: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 640 bytes in 1 blocks are still reachable in loss record 23 of 31
==22297==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x72C3D9: aligned_malloc (alloc.rs:95)
==22297==    by 0x72C3D9: alloc (alloc.rs:22)
==22297==    by 0x72C3D9: __rdl_alloc (alloc.rs:304)
==22297==    by 0x2EF342: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2EEFDD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E6922: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DE3E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2079: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E014A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2D90: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DD68E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 808 bytes in 1 blocks are still reachable in loss record 24 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x64E55F: alloc::collections::btree::map::BTreeMap<K,V>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x647071: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x286AA5: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x280D75: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x57C4EE: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x57AE63: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25BFFD: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x25D952: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x72D972: {{closure}} (rt.rs:52)
==22297==    by 0x72D972: do_call<closure-0,i32> (panicking.rs:348)
==22297==    by 0x72D972: try<i32,closure-0> (panicking.rs:325)
==22297==    by 0x72D972: catch_unwind<closure-0,i32> (panic.rs:394)
==22297==    by 0x72D972: std::rt::lang_start_internal (rt.rs:51)
==22297==    by 0x25D601: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 1,152 bytes in 4 blocks are possibly lost in loss record 25 of 31
==22297==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==22297==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==22297==    by 0x5252227: allocate_stack (allocatestack.c:627)
==22297==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==22297==    by 0x73BFEC: std::sys::unix::thread::Thread::new (thread.rs:66)
==22297==    by 0x2DD4C6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0766: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E132F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 1,520 bytes in 1 blocks are still reachable in loss record 26 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2E0D72: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2A0181: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2849E8: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 1,536 bytes in 4 blocks are still reachable in loss record 27 of 31
==22297==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x72C3D9: aligned_malloc (alloc.rs:95)
==22297==    by 0x72C3D9: alloc (alloc.rs:22)
==22297==    by 0x72C3D9: __rdl_alloc (alloc.rs:304)
==22297==    by 0x2E6820: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E7767: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0C06: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 2,008 bytes in 1 blocks are still reachable in loss record 28 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2EEA8E: crossbeam_epoch::sync::queue::Queue<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2EF312: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2EEFDD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E6922: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DE3E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2079: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E014A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2D90: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DD68E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x73C2B9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22297==    by 0x73C2B9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22297==    by 0x73C2B9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22297== 
==22297== 4,096 bytes in 4 blocks are still reachable in loss record 29 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2E6760: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E7767: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E0C06: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E44B5: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x727019: std::sync::once::Once::call_inner (once.rs:416)
==22297==    by 0x2E08E5: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E3894: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x28CC91: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x293205: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x297708: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2B3741: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297== 
==22297== 6,080 bytes in 4 blocks are still reachable in loss record 30 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2DFB18: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2D90: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DD68E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x73C2B9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22297==    by 0x73C2B9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22297==    by 0x73C2B9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22297==    by 0x52516DA: start_thread (pthread_create.c:463)
==22297==    by 0x57A271E: clone (clone.S:95)
==22297== 
==22297== 8,160 bytes in 4 blocks are still reachable in loss record 31 of 31
==22297==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22297==    by 0x2EC36F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2EEF64: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E6922: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DE3E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2079: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E014A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2E2D90: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x2DD68E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==22297==    by 0x73C2B9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22297==    by 0x73C2B9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22297==    by 0x73C2B9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22297==    by 0x52516DA: start_thread (pthread_create.c:463)
==22297==    by 0x57A271E: clone (clone.S:95)
==22297== 
==22297== LEAK SUMMARY:
==22297==    definitely lost: 0 bytes in 0 blocks
==22297==    indirectly lost: 0 bytes in 0 blocks
==22297==      possibly lost: 1,152 bytes in 4 blocks
==22297==    still reachable: 29,232 bytes in 87 blocks
==22297==         suppressed: 0 bytes in 0 blocks
==22297== 
==22297== For counts of detected and suppressed errors, rerun with: -v
==22297== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
